src/os/ui/menu/layermenu.js

```diff
@@ -367,11 +367,28 @@ os.ui.menu.layer.onFeatureList_ = function(event) {
  * @private
  */
 os.ui.menu.layer.onGoTo_ = function(event) {
-  var extent = os.ui.menu.layer.getLayersFromContext(event.getContext())
-      .reduce(os.fn.reduceExtentFromLayers, ol.extent.createEmpty());
+  // aggregate the features and execute os.feature.flyTo, in case they have altitude and pure extent wont cut it
+  const layers = os.ui.menu.layer.getLayersFromContext(event.getContext());
+  const features = layers.reduce((feats, layer) => {
+    let source = layer.getSource();
+    if (source instanceof os.source.Request) {
```

Comment:
This was probably my fault for not noticing this last time, but why is this checking for `instanceof os.source.Request` over just `ol.source.Vector`?

Reply:
Possibly a typo, but that should be `os.source.Vector` (not `ol`) given the use of `getFeatureCount`. We could support OL sources as well if we skip the `getFeatureCount` call, assign `source.getFeatures()` to something, and do the `concat` if that's defined and not empty.